File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Az_CommessaService/Models/Az_CommessaModel.cs
namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_CommessaService.Models
{
    public class Az_CommessaModel
    {
        public  int Id { get; set; }
        public  int IdAz_Anagrafica { get; set; }
        public string Descrizione { get; set; } = string.Empty;
        public  int IdAz_Cliente { get; set; }
        public Boolean Default { get; set; }
        public string Data { get; set; } = string.Empty;
        public string DataA { get; set; } = string.Empty;
    }

    public class Az_Commessa_GetAll_InModel
    {
        // Aggiungi qui eventuali parametri di filtro per la ricerca delle commesse
        // Esempio:
        // public int? IdAz_Anagrafica { get; set; }
    }

    public class Az_Commessa_GetAll_OutModel
    {
        public List<Az_CommessaModel> Az_Commessa { get; set; } = new();
    }

    public class Az_CommessaGetInModel
    {
        public int Id { get; set; } = 0;
    }

    public class Az_CommessaGetOutModel
    {
        public Az_CommessaModel Az_Commessa { get; set; } = new Az_CommessaModel();
    }

    public class Az_CommessaPutInModel
    {
        public Az_CommessaModel Az_Commessa { get; set; } = new Az_CommessaModel();
    }

    public class Az_CommessaPutOutModel
    {
        public Az_CommessaModel Az_Commessa { get; set; } = new Az_CommessaModel();
    }

    public class Az_CommessaDeleteInModel
    {
        public int Id { get; set; } = 0;
    }

    public class Az_CommessaDeleteOutModel
    {
        public Az_CommessaModel Az_Commessa { get; set; } = new Az_CommessaModel();
    }
}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Az_Commessa/Models/az-commessa-model.ts
import { CheckObjOn_Id_Number, CheckObjOn_Id_Text } from "../../../ModelsBase/check-obj";
import { ModelResult } from "../../../ModelsBase/model-result";
import { CheckObjOn_Id_Text_4ApprovalZorder } from "../../Az_SediReparto/Models/az-sedi-reparto-model";
import { Az_SubCommessa_4EditModel } from "../../Az_SubCommessa/Models/az-subcommessa-model";

export class Az_CommessaModel {
  public id!: number;
  public idAz_Anagrafica!: number;
  public descrizione: string;
  public idAz_Cliente!: number; 
  public default: boolean;
  public data!: string;
  public dataA!: string;
}

export class Az_Commessa_GetAll_InModel {}
export class Az_Commessa_GetAll_OutModel extends ModelResult {
  public az_Commessa: Az_CommessaModel[] = [];
}


export class Az_CommessaDeleteInModel {
  public id: number = 0;
}
export class Az_CommessaDeleteOutModel extends ModelResult {
  public az_Commessa!: Az_CommessaModel;
}




export class Az_CommessaGetInModel {
  public id: number = 0;
}
export class Az_CommessaGetOutModel extends ModelResult {
  public az_Commessa!: Az_CommessaModel;
  public az_SubCommessa: Az_SubCommessa_4EditModel[] = [];
  //public selected_User: CheckObjOn_Id_Text[];
  //public selected_Az_SediReparto: CheckObjOn_Id_Number[];
}

export class Az_CommessaPutInModel {
  public az_Commessa: Az_CommessaModel = new Az_CommessaModel();
  public az_SubCommessa: Az_SubCommessa_4EditModel[] = [];

  //public selected_User: CheckObjOn_Id_Text[];
  //public selected_Az_SediReparto: CheckObjOn_Id_Number[];
}
export class Az_CommessaPutOutModel extends ModelResult {
  az_Commessa!: Az_CommessaModel;
  public az_SubCommessa: Az_SubCommessa_4EditModel[] = [];
  //public selected_User: CheckObjOn_Id_Text[];
  //public selected_Az_SediReparto: CheckObjOn_Id_Number[];
}
